samples/Azure.ResourceManager.Sample/Generated/AvailabilitySetResource.Serialization.cs

```diff
@@ -0,0 +1,38 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+// <auto-generated/>
+
+#nullable disable
+
+using System;
+using System.ClientModel.Primitives;
+using System.Text.Json;
+
+namespace Azure.ResourceManager.Sample
+{
+    public partial class AvailabilitySetResource : IJsonModel<AvailabilitySetData>
+    {
+        void IJsonModel<AvailabilitySetData>.Write(Utf8JsonWriter writer, ModelReaderWriterOptions options)
+        {
+            writer.WriteStringValue(ModelReaderWriter.Write(Data, options));
```

Comment:
I think we should forward via the interface here instead of calling back into MRW.

`((IJsonModel<AvailabilitySetData>)Data).Write(writer, options);`

Reply:
updated